packages/editor/src/components/parts/mail/MailAttachmentPart.tsx

```diff
@@ -29,70 +30,77 @@ const MailAttachmentsPart = () => {
+        accessorFn: row => row.attachment,
         header: () => <span>Attachments</span>,
-        cell: cell => <ScriptCell cell={cell} type='Attachment' browsers={['attr', 'func', 'type', 'cms']} />
+        cell: cell => (
+          <ScriptCell cell={cell} type='Attachment' browsers={['attr', 'func', 'type', 'cms']} placeholder={'Enter the Attachment'} />
+        )
       }
     ],
     []
   );
 
-  const addRow = () => {
-    update('attachments', [...config.attachments, '']);
+  const onChange = (change: MailAttachment[]) => {
+    setData(change);
+    const mappedData: string[] = change.map(attachment => attachment.attachment);
+    update('attachments', mappedData);
   };
 
-  const removeTableRow = (index: number) => {
-    const newData = [...config.attachments];
-    newData.splice(index, 1);
-    update('attachments', newData);
-  };
-
-  const table = useReactTable({
-    data: config.attachments,
+  const { table, rowSelection, setRowSelection, addRow, removeRow, showAddButton } = useResizableEditableTable({
+    data,
     columns,
-    getCoreRowModel: getCoreRowModel(),
-    meta: {
-      updateData: (rowId: string, _columnId: string, value: unknown) => {
-        const rowIndex = parseInt(rowId);
-        const newData = [...config.attachments];
-        newData.splice(rowIndex, 1, value as string);
-        update('attachments', newData);
-      }
-    }
+    onChange,
+    emptyDataObject: { attachment: '' }
   });
 
+  const tableActions =
+    table.getSelectedRowModel().rows.length > 0
+      ? [
+          {
+            label: 'Remove row',
+            icon: IvyIcons.Trash,
+            action: () => removeRow(table.getRowModel().rowsById[Object.keys(rowSelection)[0]].index)
+          }
+        ]
+      : [];
+
   return (
     <>
-      <Table>
-        <thead>
-          {table.getHeaderGroups().map(headerGroup => (
-            <tr key={headerGroup.id}>
-              {headerGroup.headers.map(header => (
-                <TableHeader key={header.id} colSpan={header.colSpan}>
-                  {flexRender(header.column.columnDef.header, header.getContext())}
-                </TableHeader>
-              ))}
-              <TableHeader colSpan={1} />
-            </tr>
-          ))}
-        </thead>
-        <tbody>
-          {table.getRowModel().rows.map(row => (
-            <ValidationRow colSpan={1} key={row.id} rowPathSuffix={row.index}>
-              {row.getVisibleCells().map(cell => (
-                <TableCell key={cell.id}>{flexRender(cell.column.columnDef.cell, cell.getContext())}</TableCell>
-              ))}
-              <ActionCell actions={[{ label: 'Remove row', icon: IvyIcons.Trash, action: () => removeTableRow(row.index) }]} />
-            </ValidationRow>
-          ))}
-        </tbody>
-      </Table>
-      <TableAddRow addRow={addRow} />
+      <div>
+        {table.getSelectedRowModel().rows.length > 0 && <Fieldset label=' ' controls={tableActions} />}
```

Comment:
same here, I think it would be ok to have here a label 'Attachmets' or whatever for this fieldset

Reply:
Yes, I can do that but do you maybe know a better name for the Fieldset than "Attachments". Because I think if we set a Fieldset to Attachments the word "Attachments" is appearing too much:
![grafik](https://github.com/axonivy/inscription-client/assets/141223521/3fbe96bf-5a08-4a01-9184-26215e9440c8)

Reply:
maybe change the table header? or maybe this header is not even needed?

Reply:
Good idea, I have now simply removed the header.
![grafik](https://github.com/axonivy/inscription-client/assets/141223521/44288d65-e5ba-4ff0-95ef-9c95b6bcbdc0)